backend/code/m-booking/src/test/java/com/mbooking/service/UserServiceIntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+import com.mbooking.model.Admin;
+import com.mbooking.model.Authority;
+import com.mbooking.model.Customer;
+import com.mbooking.model.User;
+import com.mbooking.repository.AuthorityRepository;
+import com.mbooking.repository.CustomerRepository;
+import com.mbooking.repository.UserRepository;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test_h2")
+public class UserServiceIntegrationTest {
+
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private UserRepository userRepo;
+
+	@Autowired
+	private CustomerRepository customerRepo;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	@Autowired
+	private AuthorityRepository authorityRepo;
+
+	@Autowired
+	private EmailSenderService emailSenderService;    
+
+	
+	
+	/*@Before
+	public void setUp() {
+		Customer cus = new Customer();
+		cus.setEmail("email");
+		cus.setEmailConfirmationId("eid");
+		cus.setFirstname("name");
+		cus.setLastname("lastname");
+		cus.setEmailConfirmed(false);
+		cus.isEmailConfirmed();
+
+
+	       Authority roleRegistered = new Authority((long) -2,"ROLE_ADMIN");
+	        authorityRepo.save(roleRegistered);
+
+	        userRepo.save(cus);
+	      
+
+	    }*/
+		
+	
+	
+	
+	
+	@Test(expected = DataIntegrityViolationException.class)
+	public void testRegister() {
+		UserDTO userDTO = new UserDTO();
+		userDTO.setId(1L);
+		userDTO.setFirstname("user5");
+		userDTO.setLastname("user5");
+		userDTO.setEmail("email2@gmail.com");
+		userDTO.setPassword("user");
+		
+		Customer customer = new Customer();
+		customer.setEmail(userDTO.getEmail());
+		customer.setFirstname(userDTO.getFirstname());
+		customer.setLastname(userDTO.getLastname());
+		customer.setPassword(passwordEncoder.encode(userDTO.getPassword()));
+		
+		UserDTO rez = userService.register(userDTO);
+		
+		
+		emailSenderService.sendSimpleMessage(customer.getEmail(),"sub", "tex");
+		
+		
+		assertEquals(customer.getEmail(), rez.getEmail());
+		assertEquals(customer.getFirstname(), rez.getFirstname());
+		assertEquals(customer.getLastname(), rez.getLastname());
+		
+		customerRepo.save(customer);
+		authorityRepo.findByName("ROLE_CUSTOMER");
+	}
+
+	@Test(expected = ApiException.class)
+	public void testRegister1() {
+		UserDTO userDTO = new UserDTO();
+		userDTO.setId(1L);
+		userDTO.setFirstname("name");
+		userDTO.setLastname("lastname");
+		userDTO.setEmail("email");
+
+		Customer customer = new Customer();
+		customer.setEmail(userDTO.getEmail());
+		customer.setFirstname(userDTO.getFirstname());
+		customer.setLastname(userDTO.getLastname());
+
+		userService.register(userDTO);
+	}
+
+	
+
+	@Test
+	public void testCreateAdmin() {
+		UserDTO userDTO = new UserDTO();
+
+		userDTO.setId(1L);
+		userDTO.setFirstname("name");
+		userDTO.setLastname("lastname");
+		userDTO.setEmail("email");
+		userDTO.setPassword("pasw");
+		userDTO.setAuthorities(new ArrayList());
+
+		Admin admin = new Admin();
```

Comment:
The test is good, but you don't need the admin, you never use it anywhere. Make sure to remove it